include/nudb/impl/basic_store.ipp

```diff
@@ -187,6 +208,118 @@ open(
     ctx_->insert(*this);
 }
 
+template<class Hasher, class File>
+template<class... Args>
+void
+basic_store<Hasher, File>::
+open(
+    path_type const& dir_path,
+    error_code& ec,
+    Args&&... args)
+{
+  BOOST_ASSERT(boost::filesystem::exists(dir_path));
```

Comment:
This shouldn't be an assert, it should be a run-time error code, e.g. `dir_not_found`. Or perhaps we just propagate the `boost::filesystem` error if one exists.

Reply:
If we don't use `filesystem` we need to add our own error enum